spi_nand_flash/src/spi_nand_api.c

```diff
@@ -0,0 +1,353 @@
+/*
```

Comment:
Not sure if we should keep the name of this file as "spi_nand_api", mostly because it doesn't implement this component's API but rather the internal functions.

Reply:
Renamed as nand_impl.c. Please let me know for any input.